packages/orm/src/relations/ManyToManyCollection.ts

```diff
@@ -228,23 +228,28 @@ export class ManyToManyCollection<T extends Entity, U extends Entity>
       const m2m = other[this.otherFieldName] as any as ManyToManyCollection<U, T>;
       m2m.remove(this.entity);
     });
-    this.loaded = [];
+    this.#loaded = [];
   }
 
   private maybeApplyAddedAndRemovedBeforeLoaded(): void {
-    if (this.loaded) {
-      // this.loaded.unshift(...this.addedBeforeLoaded);
-      // this.addedBeforeLoaded = [];
```

Comment:
I'd noticed this before, but was not sure why these lines were commented out.

They caused my initial repro attempts to fail, but due to this bug, instead of the reactivity bug--so went ahead and fixed this while here.